pkg/sql/opt/bench/fk_test.go

```diff
@@ -53,16 +54,15 @@ func runFKBench(
 }
 
 func BenchmarkFKInsert(b *testing.B) {
+	defer log.Scope(b).Close(b)
 	const parentRows = 1000
 	setup := func(b *testing.B, r *sqlutils.SQLRunner, setupFKs bool) {
-		r.Exec(b, "CREATE TABLE child (k int primary key, p int)")
-		r.Exec(b, "CREATE TABLE parent (p int primary key, data int)")
-
+		r.Exec(b, "CREATE TABLE parent (p INT PRIMARY KEY, data INT)")
 		if setupFKs {
-			r.Exec(b, "ALTER TABLE child ADD CONSTRAINT fk FOREIGN KEY (p) REFERENCES parent(p)")
+			r.Exec(b, "CREATE TABLE child (k INT PRIMARY KEY, p INT REFERENCES parent(p))")
 		} else {
 			// Create the index on p manually so it's a more fair comparison.
-			r.Exec(b, "CREATE INDEX idx ON child(p)")
+			r.Exec(b, "CREATE TABLE child (k INT PRIMARY KEY, p INT, INDEX (p))")
```

Comment:
I made this change because, unfortunately, `defer log.Scope(b).Close(b)` does not suppress errors, and errors were being emitted by `BenchmarkFKInsert` because the test server is closed with outstanding transactions. These errors can be emitted in the middle of a benchmark output line, causing then entire benchmark case to be ignored by `benchstat`.

By creating the index in the `CREATE TABLE` statement, there is no backfill operation for the index, which I believe was causing problems for the `setupFKs=false` cases.

However, there are still errors emitted for the "NoFastPath" cases, presumably an FK check within the `INSERT` transaction is not complete when the server shuts down? I tried a few approaches for fixing it but couldn't come up with anything that worked. Any ideas?

Reply:
Looks like #59054 will fix this issue, so probably not worth spending time on.

Reply:
Removed these changes now that they're not necessary.